### PR TITLE
Add addon registry through `ember-addons` of package.json

### PR DESCRIPTION
**Pull request by suchitadoshi1987**

### CODE REVIEW

File: src/utils/layout-helpers.ts [hunk 1]
```diff
@@ -139,46 +140,90 @@ export function isELSAddonRoot(root: string) {
...
-    globs: ['**/package.json'],
-  });
-  const roots: string[] = [];
+function getRecursiveInRepoAddonRoots(root: string, roots: string[]) {
+  const packageData = getPackageJSON(root);
+  const emberAddonPaths: string[] = (packageData['ember-addon'] && packageData['ember-addon'].paths) || [];
+
+  if (roots.length) {
+    if (!isEmberAddon(packageData)) {
+      return [];
```
Comment (lifeart) [inline src/utils/layout-helpers.ts:149]: should we return roots here?

File: src/utils/layout-helpers.ts [hunk 2]
```diff
@@ -139,46 +140,90 @@ export function isELSAddonRoot(root: string) {
...
-          roots.push(relatedRoot);
-        }
+  return recursiveRoots;
+}
+
+export function getProjectInRepoAddonsRoots(root: string) {
+  const roots: string[] = [];
+
+  if (!isModuleUnificationApp(root)) {
+    ['lib', 'engines'].forEach((prefix) => {
```
Comment (lifeart) [inline src/utils/layout-helpers.ts]: looke like we don't need it for nor module unification app, it should be like:

```js
getProjectInRepoAddonsRoots(root) {
   if (isModuleUnificationApp(root)) {
     // ...legacy logic
   } else {
     // new logic, based on package.json 
   }
}

```

currently:

```js
if (not(moduleUnification(root)) { 
 // legacy logic
 }
```

File: src/utils/layout-helpers.ts (hunk 1 above)
Comment (suchitadoshi1987) [reply src/utils/layout-helpers.ts:149]: i dont think so, becauseif the packageData is not an addon, then there shouldnt be any roots returned

File: src/utils/layout-helpers.ts (hunk 2 above)
Comment (suchitadoshi1987) [reply src/utils/layout-helpers.ts]: nope, we need to iterate over the `lib` and `engines` of the package paths to get the `engine-addons` of addons. Else we will only leverage whatever is in the root's `engine-addons` which might not be a complete list. If you see carefully, the `not(MOduleUnififcation)` code has the logic based on package json

**Review by lifeart — APPROVED** [review]

**Pull request merged**